src/Drivers/nlpPriDec_ex8_driver.cpp

```diff
@@ -0,0 +1,179 @@
+ * 
+ * @note This example is built only when HIOP_USE_MPI is enabled during cmake build
+ * and require at least two MPI ranks in MPI_COMM_WORLD.
+ *
+ */
+
+static bool self_check(int nx, int S, double obj_value);
+static bool parse_arguments(int argc, char **argv,
+			    bool& self_check,
+			    int& nx,
+			    int& S)
+{
+  self_check = false;
+  nx = 20;
+  S = 100;
+
+  switch(argc) {
+  case 1:
+    //no arguments
+    return true;
+    break;
+  case 4: // 3 arguments
+    {
+      if(std::string(argv[3]) == "-selfcheck")
+      {
+	self_check=true;
+        nx = std::atoi(argv[1]);
+        S = std::atof(argv[2]);
+        if(S<3) S = 4;
+        if(nx<=0) return false;
+      } else {
+        return false;
+      }
+    }
+  case 3: //2 arguments
+    {
+      nx = atoi(argv[1]);
+      if(nx<=0) return false;
+      S = atoi(argv[2]);
+      if(S<3) S = 4;
+    }
+  case 2: //1 argument
+    {
+      if(std::string(argv[1]) == "-selfcheck")
+      {
+        self_check=true;
+      } else {
+        nx = atoi(argv[1]);
+        if(nx<=0) return false;
+      }
+    }
+    break;
+  default: 
+    return false; //4 or more arguments
+  }
+
+  if(self_check && nx!=20 && S!=100) {
+      printf("Error: incorrect input parameters: '-selfcheck' must be used with predefined "
+	     "values for input  parameters, nx=20 S=100.\n");
+      return false;
+  }
+  
+  return true;
+};
+
+static void usage(const char* exeName)
+{
+  printf("HiOp driver %s that solves a nonconvex synthetic problem of variable size in the "
+	 "primal decomposition formulation. )\n", 
+	 exeName);
+  printf("Usage: \n");
+  printf("  '$ %s nx S -selfcheck '\n", exeName);
+  printf("Arguments, all integers, excepting strings '-selfcheck' \n");
+  printf("  'nx': # of base case variables [default 20, optional, nonnegative integer].\n");
+  printf("  'S': # of recourse/contingency problems [default 100, optional, nonnegative integer].\n");
+  printf("  '-selfcheck': compares the optimal objective with nx being 20 and "
+	 "S being 100 (these two exact values must be passed as arguments). [optional]\n");
+}
+
+
+int main(int argc, char **argv)
+{
+  int rank=0;
+#ifdef HIOP_USE_MPI
+  MPI_Init(&argc, &argv);
+  int comm_size;
+  int ierr = MPI_Comm_size(MPI_COMM_WORLD, &comm_size); assert(MPI_SUCCESS==ierr);
+  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank); assert(MPI_SUCCESS==ierr);
+#endif
+
+#ifdef HIOP_USE_MAGMA
+  magma_init();
+#endif
+  int nx=20;
+  int S=100;
+  int nc=20;
+  //index of coupled x in the entire x
+  //for testing purpos
+  std::vector<int> list(nc);
+  for(int i=0;i<nc;i++) list[i]=i;
+ 
+  bool selfCheck;
+  
+  if(!parse_arguments(argc, argv, selfCheck, nx, S)) {
+    usage(argv[0]);
+    return 1;
+  }
+
+  //PriDecMasterProblemEx8 pridec_problem(nx, S,MPI_COMM_WORLD);
+  PriDecMasterProblemEx8 pridec_problem(nx, S, nc,MPI_COMM_WORLD);
+  //printf("total ranks %d\n",comm_size);
```

Comment:
I suggest removing commented out code before the merge.
```suggestion
  PriDecMasterProblemEx8 pridec_problem(nx, S, nc, MPI_COMM_WORLD);
```